coordinator/internal/transitengineapi/transitengineapi.go

```diff
@@ -51,6 +53,39 @@ type stateAuthority interface {
 	GetState() (*authority.State, error)
 }
 
+// NewTransitEngineAPI sets up the transit engine API with a provided seedEngineAuthority.
+func NewTransitEngineAPI(authority stateAuthority, port int, logger *slog.Logger) *http.Server {
+	return &http.Server{
+		Addr: fmt.Sprintf(":%d", port),
+		TLSConfig: &tls.Config{
+			ClientAuth: tls.RequireAndVerifyClientCert,
+			GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+				logger.Debug("call getConfigForClient")
+				state, err := authority.GetState()
+				if err != nil {
+					return nil, fmt.Errorf("getting state: %w", err)
+				}
+				if len(state.CA.GetMeshCACert()) == 0 {
+					return nil, fmt.Errorf("mesh ca cert not initialized")
+				}
```

Comment:
Can this even happen? I'd say that an initialized CA should always have a mesh cert.